File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/common/client/jaxrs/CustomPoolingHttpClientConnectionManager.java
package co.uk.swarmbit.docker.api.common.client.jaxrs;

import co.uk.swarmbit.docker.config.DockerClientConfig;
import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;

public class CustomPoolingHttpClientConnectionManager extends PoolingHttpClientConnectionManager {

    public CustomPoolingHttpClientConnectionManager(DockerClientConfig config, org.apache.http.config.Registry registry) {
        super(registry);
        if (config.getMaxTotalConnections().isPresent()) {
            this.setMaxTotal(config.getMaxTotalConnections().get());
        }
        if (config.getMaxTotalConnections().isPresent()) {
            this.setMaxTotal(config.getMaxTotalConnections().get());
        }
    }
    @Override
    public void close() {
        super.shutdown();
    }

    @Override
    public void shutdown() {
        // Disable shutdown of the pool. This will be done later, when this factory is closed
        // This is a workaround for finalize method on jerseys ClientRuntime which
        // closes the client and shuts down the connection pool when it is garbage collected
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/client/jaxrs/CustomPoolingHttpClientConnectionManager.kt
package com.swarmbit.docker.api.client.jaxrs

import com.swarmbit.docker.api.config.DockerClientConfig
import org.apache.http.config.Registry
import org.apache.http.conn.socket.ConnectionSocketFactory
import org.apache.http.impl.conn.PoolingHttpClientConnectionManager

class CustomPoolingHttpClientConnectionManager(
        config: DockerClientConfig,
        registry: Registry<ConnectionSocketFactory>
) : PoolingHttpClientConnectionManager(registry) {

    override fun close() {
        super.shutdown()
    }

    override fun shutdown() {
        // Disable shutdown of the pool. This will be done later, when this factory is closed
        // This is a workaround for finalize method on jerseys ClientRuntime which
        // closes the client and shuts down the connection pool when it is garbage collected
    }

    init {
        config.maxTotalConnections?.let {
            maxTotal = it
        }
        config.maxPerRouteConnections?.let {
            defaultMaxPerRoute = it
        }
    }
}